presto-main/src/test/java/io/prestosql/sql/TestExpressionInterpreter.java

```diff
@@ -1132,16 +1132,16 @@ public void testSimpleCase()
     @Test
     public void testCoalesce()
     {
-        assertOptimizedEquals("coalesce(2 * 3 * unbound_long, 1 - 1, null)", "coalesce(6 * unbound_long, 0)");
-        assertOptimizedEquals("coalesce(2 * 3 * unbound_long, 1.0E0/2.0E0, null)", "coalesce(6 * unbound_long, 0.5E0)");
+        assertOptimizedEquals("coalesce(unbound_long * (2 * 3), 1 - 1, null)", "coalesce(unbound_long * 6, 0)");
```

Comment:
why did you change the input too (`coalesce(2 * 3 * unbound_long, 1 - 1, null)`) ?

Reply:
I think constant folding is happening after canonicalization in some cases, so they weren't matching. I haven't investigated further, though.

Reply:
Oh, yeah, `assertOptimizedEquals` is optimizing the expression directly.